src/api/app/models/workflow.rb

```diff
@@ -135,8 +135,19 @@ def destroy_target_projects
     EventSubscription.where(channel: 'scm', token: token, package: target_packages).delete_all
 
     target_project_names = processable_steps.map(&:target_project_name).uniq.compact
-    # We want the callbacks to run after destroy, so we can't use delete_all
-    Project.where(name: target_project_names).destroy_all
+
+    results = []
+    Project.where(name: target_project_names).find_each do |project|
+      begin
+        project.destroy
+      rescue Project::Errors::UnknownObjectError
+        results.append("Project not found: #{project.name}")
+      rescue Pundit::NotAuthorizedError
+        results.append("Permission error for #{project.name}")
```

Comment:
If you do not let the exception bubble up then the request will not fail and it'll show up as working in the SCM UI.